### PR TITLE
Feat ✨: Local Stream Injection

### PR DESCRIPTION
**Pull request by Yogesh-Dubey-Ayesavi**
### Summary of Changes

This pull request introduces new methods to the `JanusPlugin` class to handle the disposal of remote streams while retaining the local stream under certain scenarios. The new methods are:

- [`hangupWithoutLocalStreamDisposing`](https://github.com/Ayesavi/flutter_janus_client/blob/ae549741423f682dde96cee3811b341ffec4c58f/lib/janus_plugin.dart#L300): A method to dispose only the remote stream, preserving the local stream.

- [`injectMedia`](https://github.com/Ayesavi/flutter_janus_client/blob/2a15884f9a306b1c031a8657859826afb6680f57/lib/janus_plugin.dart#L390): A method to inject a local stream from external sources.

- [`disposeOnlyRemoteStreamWithPlugin`](https://github.com/Ayesavi/flutter_janus_client/blob/2a15884f9a306b1c031a8657859826afb6680f57/lib/janus_plugin.dart#L370): A method to dispose only the remote stream when a local stream is injected.

#### 1. Preserving Local Stream after Hangup

There are scenarios where it's necessary to retain the local stream even after the call has been hung up. This is particularly useful when users want to apply filters or perform post-processing on the local stream. The method `hangupWithoutLocalStreamDisposing` facilitates this use case by disposing only the remote stream, leaving the local stream intact.

```dart
// Example of using hangupWithoutLocalStreamDisposing
janusPlugin.hangupWithoutLocalStreamDisposing();
```

#### 2. Injecting Local Stream from External Sources

In certain situations, users may want to inject a local stream from external sources, such as after applying filters or modifications. The method `injectMedia` allows users to inject a local stream into the plugin for further processing or streaming.

```dart
// Example of using injectMedia
MediaStream localStream = // Obtain local stream from external source
janusPlugin.injectMedia(localStream);
```

#### 3. Disposing Only Remote Stream with Injected Local Stream

When a local stream is injected into the plugin, and the user decides to dispose of the streams, it's essential to use `disposeOnlyRemoteStreamWithPlugin` to ensure that only the remote stream is disposed, leaving the injected local stream unaffected.

```dart
// Example of using disposeOnlyRemoteStreamWithPlugin
janusPlugin.disposeOnlyRemoteStreamWithPlugin();
```

#### 4. Reconnecting without Disposing Local Stream
In scenarios where a reconnecting feature is implemented, it's common not to dispose of the local stream during the reconnection process. This is crucial to maintain a seamless user experience without the need to reacquire and inject the local stream. The existing method hangupWithoutLocalStreamDisposing can be particularly useful in this context, ensuring that only the remote stream is disposed while keeping the local stream intact.

### CODE REVIEW

**Review by shivanshtalwar0 — CHANGES_REQUESTED** [review]

File: lib/wrapper_plugins/janus_video_room_plugin.dart [hunk 1]
```diff
@@ -304,6 +304,12 @@ class JanusVideoRoomPlugin extends JanusPlugin {
     await this.send(data: {"request": "leave"});
   }
 
+  /// sends hangup request on current  active [JanusVideoRoomPlugin] to tear off active PeerConnection in-effect leaving the room. without disposing localstream
+  Future<void> hangupWithoutDisposing() async {
+    await super.hangupWithoutLocalStreamDisposing();
+    await this.send(data: {"request": "leave"});
```
[the remark this answers is not in the document]
Comment (Yogesh-Dubey-Ayesavi) [reply lib/wrapper_plugins/janus_video_room_plugin.dart:310]: Yes, one can create a new peer connection and attach plugin and inject local media stream

File: lib/janus_plugin.dart [hunk 2]
```diff
@@ -296,6 +296,12 @@ class JanusPlugin {
     }
   }
 
+  /// hangs up the call Diposes remote media streams only
+  Future<void> hangupWithoutLocalStreamDisposing() async {
```
[the remark this answers is not in the document]
Comment (Yogesh-Dubey-Ayesavi) [reply lib/janus_plugin.dart:300]: i think i should remove cancel polling timer

File: lib/janus_plugin.dart [hunk 3]
```diff
@@ -351,6 +366,59 @@ class JanusPlugin {
...
+    _renegotiationNeededController?.close();
+    _wsStreamSubscription?.cancel();
+
+    await webRTCHandle?.peerConnection?.close();
+    await webRTCHandle?.remoteStream?.dispose();
+    await webRTCHandle?.peerConnection?.dispose();
+  }
+
+  /// Call this method when you want to indect localstream from outside
+  Future<MediaStream?> injectMedia(
```
[the remark this answers is not in the document]
Comment (Yogesh-Dubey-Ayesavi) [reply lib/janus_plugin.dart:390]: I don't think of any side effects?

[the remark this answers is not in the document]
Comment (shivanshtalwar0) [reply lib/janus_plugin.dart:390]: i meant clearly state in dart docs what will happen on calling this function

File: lib/wrapper_plugins/janus_video_room_plugin.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (shivanshtalwar0) [reply lib/wrapper_plugins/janus_video_room_plugin.dart:310]: but you do realise when we want to establish communication again we need to attach a new plugin that means this is of no relevance to previous plugin instance

**Pull request closed without merging**